core/templates/components/state-editor/state-editor-properties-services/state-name.service.ts

```diff
@@ -23,7 +23,7 @@ import { Injectable } from '@angular/core';
 })
 export class StateNameService {
   private stateNameEditorIsShown: boolean = false;
-  private savedMemento: string = null;
+  private savedMemento: string | null = null;
```

Comment:
Why `null` needs to be added? Is there a solution where `null` won't be needed?

Reply:
`null` is being passed here
```typescript
  init(): void {
    this.setStateNameSavedMemento(null);
    this.setStateNameEditorVisibility(false);
  }
```
Another solution is to assign an empty string to `savedMemento` instead of `null`; I checked the files calling StateNameService and using just a string should not break the code but inside the test file `state-name.service.spec.ts` it expects null values of `savedMemento`. This will have to be changed if we want a solution where `null` won't be needed.
```typescript
expect(sns.getStateNameSavedMemento()).toBeNull();
```

Reply:
Okay, lets keep it like this for now.

Reply:
Done